plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/dlf/DlfMetaStoreClient.java

```diff
@@ -0,0 +1,1538 @@
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HdfsContext hdfsContext;
+    private final Optional<String> defaultDir;
+    private final IDataLakeMetaStore dataLakeMetaStore;
+    private short pageSize;
+    private int batchSizeForGetPartititon;
+    private boolean isProxyModeDlfOnly;
+    private boolean enableFsOperation;
+
+    @Inject
+    public DlfMetaStoreClient(
+            HdfsEnvironment hdfsEnvironment,
+            DlfMetaStoreConfig dlfConfig) throws TrinoException
+    {
+        logger.info("DlfMetaStoreClient start");
+        requireNonNull(dlfConfig, "dlfConfig is null");
+
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.hdfsContext = new HdfsContext(ConnectorIdentity.ofUser(DEFAULT_METASTORE_USER));
+
+        this.dlfConfig = dlfConfig;
+        this.catalogId = DataLakeConfig.DEFAULT_CATALOG_ID;
+        this.pageSize = dlfConfig.getPageSize();
+        this.defaultDir = dlfConfig.getDefaultWarehouseDir();
+        this.dataLakeMetaStore = new MetastoreFactory().getMetaStore(dlfConfig);
+        this.batchSizeForGetPartititon = dlfConfig.getBatchSizeForGetPartititon().orElse(DataLakeConfig.DEFAULT_CATALOG_COL_STATS_PAGE_SIZE);
+
+        this.isProxyModeDlfOnly = dlfConfig.getProxyMode().orElse(ProxyMode.DLF_ONLY) == ProxyMode.DLF_ONLY;
+        this.enableFsOperation = this.isProxyModeDlfOnly || (dlfConfig.getEnableFileOperation().orElse(false) && DataLakeUtil.isEnableFileOperationGray(dlfConfig.getEnableFileOperationGrayRate().orElse(0.0d)));
+        logger.info("DlfMetaStoreClient end");
+        //reInitDataLakeClient(dlfConfig);
+    }
+
+    private static List<String> buildPartitionNames(List<Column> partitionColumns, List<Partition> partitions)
+    {
+        return partitions.stream()
+                .map(partition -> makePartitionName(partitionColumns, partition.getValues()))
+                .collect(Collectors.toList());
+    }
+
+    public IDataLakeMetaStore getDataLakeMetaStore()
+    {
+        return dataLakeMetaStore;
+    }
+
+    @Override
+    public Optional<Database> getDatabase(String databaseName)
+    {
+        return getDatabase(catalogId, databaseName);
+    }
+
+    public Optional<Database> getDatabase(String catalogId, String databaseName)
+    {
+        try {
+            try {
+                com.aliyun.datalake20200710.models.Database database = this.dataLakeMetaStore.getDatabase(catalogId, databaseName);
+                return Optional.of(CatalogToTrinoConverter.toTrinoDatabase(database));
+            }
+            catch (DataLakeAPIException e) {
+                throw CatalogToTrinoConverter.toTrinoException(e.getResult(), e.getAction(), e);
+            }
+        }
+        catch (NoSuchObjectException e) {
+            return Optional.empty();
+        }
+        catch (Exception e) {
+            String msg = "unable to get database:";
+            logger.error(msg, e);
```

Comment:
Please remove the logging. We either log, or throw, but not both. If you do both, the same exception appears multiple times in the log, which is confusing.